src/sidebar/services/toast-messenger.ts

```diff
@@ -60,19 +92,29 @@ export class ToastMessengerService {
    * `type` and `message` text of an existing message).
    */
   private _addMessage(
-    type: 'error' | 'success' | 'notice',
+    type: MessageType,
     messageText: string,
     {
       autoDismiss = true,
       moreInfoURL = '',
       visuallyHidden = false,
+      delayed = false,
     }: MessageOptions = {}
   ) {
     // Do not add duplicate messages (messages with the same type and text)
     if (this._store.hasToastMessage(type, messageText)) {
       return;
     }
 
+    // Delayed messages should just be enqueued when the document is not focused
+    if (delayed && !this._window.document.hasFocus()) {
```

Comment:
What do you mean? Are you asking if messages delayed while the tab is hidden will be properly pushed when the tab becomes active, or if we should be delaying messages at all while the tab is hidden?

Reply:
I was just wondering aloud how tab visibility would affect this. For the initial use case of reporting when a highlight is created it doesn't actually matter since the tab will (almost?) definitely be visible when the highlight is created.